### ISSUE TITLE
Error uploading to pypi [twine]

### ISSUE THREAD

**Issue by duncanphillips** [travis-ci/dpl#296]
It looks like an issue has been introduced in #253 

we are now getting this error on our builds

```
Installed /usr/local/lib/python2.7/dist-packages/setuptools-18.1-py2.7.egg
Processing dependencies for setuptools==18.1
Finished processing dependencies for setuptools==18.1
pip install --user twine
You are using pip version 6.0.7, however version 7.1.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
```

**Comment by duerrp** [comment]
I get the same error on mine...
```
1.74s
dpl.1

Installing deploy dependencies

sudo: must be setuid root

--2015-08-09 05:14:01--  https://bootstrap.pypa.io/ez_setup.py

Resolving bootstrap.pypa.io (bootstrap.pypa.io)... 23.235.39.175

Connecting to bootstrap.pypa.io (bootstrap.pypa.io)|23.235.39.175|:443... connected.

HTTP request sent, awaiting response... 200 OK

Length: 11432 (11K) [text/x-python]

Saving to: `STDOUT'

 0% [                                       ] 0           --.-K/s   in 0s      

Cannot write to `-' (Success).

pip install --user twine

You are using pip version 6.0.7, however version 7.1.0 is available.

You should consider upgrading via the 'pip install --upgrade pip' command.

Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
```

**Comment by dangra** [comment]
any workaround that doesn't require sudo?

**Comment by dangra** [comment]
A simple workaround is to install `twine` in `install` section like https://github.com/scrapy/parsel/commit/380841963e12cacff0222185b1efe6dddd0bed3f#diff-354f30a63fb0907d4ad57269548329e3R9

**Comment by dhermes** [comment]
Notice #299 is the same root issue.

**Comment by BanzaiMan** [comment]
I pushed a commit to install `twine` without the `--user` flag. Could any of you try deployment with edge: true, but without installing twine prior to the deployment?

**Comment by desbma** [comment]
@BanzaiMan adding `edge: true` fixes it for me.

**Comment by BanzaiMan** [comment]
Great! I've released 1.7.19. `edge: true` is no longer necessary.
